flink-connector/docs/content/docs/connectors/datastream/pubsub.md

```diff
@@ -97,8 +104,20 @@ balances message delivery across readers.
 [`PubsubMessage`](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage)
 is deserialized into an output that is suitable for processing.
 
-For convenience, `PubSubDeserializationSchema.dataOnly` can be used if only the
-field `PubsubMessage.data` is required for processing.
+For convenience,
+`PubSubDeserializationSchema.dataOnly(DeserializationSchema<T> schema)` can be
+used if the field `PubsubMessage.data` stores all of the data to be processed.
+Additionally, Flink provides basic schemas for further convenience.
+
+```java
+// Deserialization output is PubsubMessage.data field converted to String type.
+PubSubSource.<String>builder()
+    .setDeserializationSchema(PubSubDeserializationSchema.dataOnly(new SimpleStringSchema()))
+```
+
+Implementing `PubSubDeserializationSchema<T>` is required to process data stored
+in fields other than `PubsubMessage.data`.
+<!-- TODO(matt-kwong): Add custom deserialization code example. -->
```

Comment:
is this TODO supposed to still be here?

Reply:
it's fine if it is; just checking

Reply:
Yes, I wrote some sample code to illustrate how to serialize/deserialize data to/from the PubsubMessage.attributes field, but it was too long to include in documentation IMO. I plan to create another Flink job example that focuses on serialization/deserialization.